### PR TITLE
datapath: ICMP CT fixes

### PR DESCRIPTION
**Pull request by brb**
See commit msgs.

The PR has been previously reviewed by @kkourt and @borkmann .

```release-note
Fix ICMP Echo ID placement in CT maps
```

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Review by kkourt — APPROVED** [review]

**Comment by brb** [comment]
4.19 is hitting complexity issues.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-4.9

**Comment by brb** [comment]
Converted to draft until the complexity issue has been resolved.

**Comment by brb** [comment]
The complexity issue should be resolved by https://github.com/cilium/cilium/pull/15217.

**Comment by pchaigno** [comment]
test-1.19-4.19

**Comment by jibi** [comment]
test-me-please

**Comment by jibi** [comment]
retest-gke

**Comment by jibi** [comment]
test-me-please

**Comment by jibi** [comment]
retest-runtime

**Comment by jibi** [comment]
test-me-please

**Comment by jibi** [comment]
retest-netnext

**Comment by jibi** [comment]
retest-net-next

**Comment by jibi** [comment]
Looks like net-next is hitting https://github.com/cilium/cilium/issues/15737, marking as ready to merge.

File: bpf/lib/conntrack.h [hunk 1]
```diff
@@ -754,6 +754,7 @@ static __always_inline int ct_create6(const void *map_main, const void *map_rela
 
 	entry.lb_loopback = ct_state->loopback;
 	entry.node_port = ct_state->node_port;
+	relax_verifier();
```
Comment (brb) [inline bpf/lib/conntrack.h:757]: Could you add a comment saying why it's needed?

Comment (jibi) [reply bpf/lib/conntrack.h:757]: We don't have such comments for all the other points where we added a `relax_verifier()` pruning point (as the context for that is in the commit that introduced back this helper) so I'd rather not do that for consistency.

I could add some context on how I found these pruning points, but that would probably be too long (I have a WIP document where I describe the process which is already a few pages long and I'm working on some scripts to automate this a bit more)

Comment (brb) [reply bpf/lib/conntrack.h:757]: Could you then add an explanation of your idea to the git commit message?

**Comment by jibi** [comment]
No need to rerun full CI, I just amended the latest commit message

File: bpf/lib/conntrack.h (hunk 1 above)
Comment (brb) [reply bpf/lib/conntrack.h:757]: Thanks!

**Comment by brb** [comment]
Marking it as ready-to-merge.

**Review by pchaigno — APPROVED** [review]

File: bpf/lib/conntrack.h [hunk 2]
```diff
@@ -845,6 +846,7 @@ static __always_inline int ct_create4(const void *map_main,
...
```
Comment (borkmann) [inline bpf/lib/conntrack.h:849]: Btw, do we have a cheaper call for relax_verifier() internally with less overhead and which could potentially be inlined?

Comment (pchaigno) [reply bpf/lib/conntrack.h:849]: :thinking: What do you have in mind? We also want something that has zero arguments to minimize impact on complexity.

**Pull request merged**